src/shared/stores/data-value-store.js

```diff
@@ -0,0 +1,100 @@
+import create from 'zustand'
+import { CAN_HAVE_LIMITS_TYPES } from '../value-types.js'
+
+export const useStore = create((set, get) => ({
+    highlightedField: null,
+    dataValueSet: {
+        dataValues: {},
+        minMaxValues: [],
+    },
+    getDataValue: ({ dataElementId, categoryOptionComboId }) =>
+        get().dataValueSet.dataValues[dataElementId]?.[categoryOptionComboId],
+    getDataElement: ({ dataElementId }) =>
+        get().dataValueSet.dataValues[dataElementId],
+    getDataValues: () => get().dataValueSet?.dataValues,
+    getDataValueSet: () => get().dataValueSet,
+    setDataValueSet: (values) => set({ ...get(), dataValueSet: values ?? {} }),
+    getMinMaxValues: ({ dataElementId, categoryOptionComboId }) => {
+        return get().dataValueSet?.minMaxValues?.find(
+            (minMaxValue) =>
+                minMaxValue.categoryOptionCombo === categoryOptionComboId &&
+                minMaxValue.dataElement === dataElementId
+        )
+    },
+    getHighlightedField: () => get().highlightedField,
+    isFieldHighlighted: ({ dataElementId, categoryOptionComboId }) =>
+        get().highlightedField?.de?.id === dataElementId &&
+        get().highlightedField?.coc?.id === categoryOptionComboId,
+    setHighlightedField: (value) => {
+        set({ ...get(), highlightedField: value ?? null })
+    },
+    getHighlightedFieldData: () => {
+        const highlightedField = get().getHighlightedField()
```

Comment:
Having a centralised store means that you can have more complex selectors - I think there are a couple of places where we "munge" the highlighted item data and data value so potentially you could do this in the selector

Reply:
I think we should keep them separate, and let the `dataValue`-store be an exact reference to the same object as RQ-return data.

Reply:
Ok I'll drop the commits for HighlightedField